### ISSUE TITLE
X11 builds with native compilation don't work

### ISSUE THREAD

**Issue by chetstone** [d12frosted/homebrew-emacs-plus#360]
Builds OK with `brew install emacs-plus@28 --with-x11 --with-retro-sink-bw-icon --with-native-comp` but when run it fails with the same error message as #333:

```
emacs: can't find function "F6d656e752d6261722d7570646174652d62756666657273_menu_bar_update_buffers_0" in compilation unit /usr/local/Cellar/emacs-plus@28/28.0.50/libexec/emacs/28.0.50/x86_64-apple-darwin20.4.0/../../../../bin/../lib/emacs/28.0.50/native-lisp/28.0.50-790491c2/preloaded/menu-bar-4f46ea94-bb2650c8.eln
```
 I've tried [pinning the build](#357) to known good commits, including [3f8b303](https://github.com/jimeh/emacs-builds/issues/7) and f4dc646e0d7fb673f3149836bb7299fba9539e80, which I used to build a cocoa version with native compilation about 10 days ago and has been working well. All fail with the same error.

The X11 builds work fine if --with-native-comp is not used.

**Comment by d12frosted** [comment]
Seems like something is wrong with menu bar. I also see it in https://github.com/d12frosted/homebrew-emacs-plus/issues/361

> f4dc646e0d7fb673f3149836bb7299fba9539e80, which I used to build a cocoa version with native compilation about 10 days ago and has been working well. All fail with the same error.

Sorry, didn't understand. Doesn't that mean that f4dc646e0d7fb673f3149836bb7299fba9539e80 is used to work, but is not working anymore? Did you update macos in those 10 days?

I am using a pretty old macbooks (2014 and 2009), so don't have this issue.

**Comment by chetstone** [comment]
No,  f4dc646e0d7fb673f3149836bb7299fba9539e80  works well when built with cocoa and native compilation. I've been using it for a couple of weeks.

 But it's one of several commits I used to try to build with X11 and native compilation, and none of them worked--- all failed with the same error.

**Comment by thinkiny** [comment]
Same issue here，i use ```brew install emacs-plus@28 --with-xwidgets --with-modern-cg433n-icon --with-native-com```

**Comment by mkvoya** [comment]
Same issue here, too. This issue seems to be the same as #364 .
In the output of `objdump -D menu-bar-4f46ea94-7781c24c.eln`, I can see a function named `_F6d656e752d6261722d7570646174652d62756666657273_menu_bar_update_buffers_0`. Note the leading underscore. I doubt the issue is caused by the extra leading underscore.

**Comment by d12frosted** [comment]
Should be fixed via 4f44f33. Please verify.
